windows/check_quiesce_snapshot/check_quiescing_type.yml

```diff
@@ -0,0 +1,88 @@
+
+- name: "Set fact of the list of vss_manifests zip files"
+  ansible.builtin.set_fact:
+    vm_vss_manifests_list_after: "{{ vm_guest_facts.instance.hw_files | select('search', 'vss_manifests') }}"
+
+- name: "Display the vss_manifests zip files after taking quiesce snapshot"
+  ansible.builtin.debug: var=vm_vss_manifests_list_after
+
+- name: "Set fact of the new vss_manifests zip file"
+  ansible.builtin.set_fact:
+    vm_vss_manifests_new: "{{ vm_vss_manifests_list_after | difference(vm_vss_manifests_list_before) }}"
+
+- name: "Check one new vss_manifests zip file generated"
+  ansible.builtin.assert:
+    that:
+      - vm_vss_manifests_new | length == 1
+    fail_msg: >
+      Before taking quiesce snapshot, the vss_manifests zip files are: {{ vm_vss_manifests_list_before }},
+      after taking quiesce snapshot, the vss_manifests zip files are: {{ vm_vss_manifests_list_after }},
+      not get the new generated zip file.
+
+- name: "Set fact of local path of the new vss_manifests zip file"
+  ansible.builtin.set_fact:
+    vm_vss_manifests_path: "{{ current_test_log_folder }}/{{ vm_vss_manifests_new[0] | basename }}"
+    vm_vss_manifests_unzip_path: "{{ current_test_log_folder }}/{{ (vm_vss_manifests_new[0] | basename).split('.')[0] }}"
+
+- name: "Create local directory for unarchived files"
+  include_tasks: ../../common/create_directory.yml
+  vars:
+    dir_path: "{{ vm_vss_manifests_unzip_path }}"
+
+- name: "Get the new vss_manifests zip file to local"
+  include_tasks: ../../common/esxi_download_datastore_file.yml
+  vars:
+    src_datastore: "{{ datastore }}"
+    src_file_path: "{{ vm_vss_manifests_new[0].split(']')[1].strip() }}"
+    dest_file_path: "{{ vm_vss_manifests_path }}"
+
+- name: "Get the list of files in the new vss_manifests zip file"
+  ansible.builtin.unarchive:
+    src: "{{ vm_vss_manifests_new_path }}"
+    dest: "{{ vm_vss_manifests_unzip_path }}"
+    list_files: true
+  register: vm_vss_manifests_files
+
+- name: "Set fact of not support application consistent quiescing situation"
+  ansible.builtin.set_fact:
+    winsrv_app_quiesce_support: true
+  when:
+    - guestinfo_guest_full_name | regex_search('Windows Server ([0-9]{4})')
+    - guestinfo_guest_full_name | regex_search('Windows Server ([0-9]{4})') == guest_os_ansible_distribution | regex_search('Windows Server ([0-9]{4})')
+
+- name: "Check filesystem consistent quiescing taken"
+  ansible.builtin.assert:
+    that:
+      - vm_vss_manifests_files.files is defined
+      - vm_vss_manifests_files.files | length == 1
+      - vm_vss_manifests_files.files[0] == "backup.xml"
+    fail_msg: >
+      One file 'backup.xml' is expected in {{ vm_vss_manifests_new[0] | basename }} when
+      taking filesystem consistent quiesce snapshot, while got files: {{ vm_vss_manifests_files.files | default([]) }}.
+  when: guest_os_product_type == 'client' or not winsrv_app_quiesce_support
```

Comment:
From above set_fact task, only Windows Server could have `winsrv_app_quiesce_support=true`, which means if the guest OS is Windows Client, its `winsrv_app_quiesce_support` would always be false. So this when can only use `not winsrv_app_quiesce_support`.

Reply:
Yeah, right, thanks.